ci/packaging/suse/rpmfiles_maker.sh

```diff
@@ -1,40 +1,81 @@
 #!/bin/bash
+##########################################
+# generate skuba tar.gz and specfile
+##########################################
 
+##########################################
+# Housekeeping
 set -e
-
-susepkg_dir=$(cd "$( dirname "$0" )" && pwd)
 tmp_dir=$(mktemp -d -t skuba_XXXX)
-rpm_files="${susepkg_dir}/obs_files"
-devel_prj="Devel:CaaSP:5"
-version="$1"
-tag="$2"
-closest_tag="$3"
-
 log()   { (>&2 echo ">>> $*") ; }
 clean() { log "Cleaning temporary directory ${tmp_dir}"; rm -rf "${tmp_dir}"; }
+trap clean ERR
+susepkg_dir=$(cd "$( dirname "$0" )" && pwd)
+rpm_files="${susepkg_dir}/obs_files"
+devel_prj="Devel:CaaSP:5"
 
-if [ $# -ne 3 ]; then
+##########################################
+# CLI args
+if (( $# != 3 )); then
     log "usage: <version> <tag> <closest-tag>"
     exit 1
 fi
+version="$1"     # pretty version string for display 
+tag="$2"         # can be ugly refs/tags/xxx or specific commit hash
+closest_tag="$3" # usually $tag, but might be commit
 
-trap clean ERR
+##########################################
+# env vars which come from GitHub Actions
+TARBALL_PATH=${TARBALL_PATH:-${tmp_dir}/skuba.tar.gz}
+TEMPLATE_PATH=${TEMPLATE_PATH:-${susepkg_dir}/skuba_spec_template}
+SPECFILE_PATH=${SPECFILE_PATH:-${tmp_dir}/skuba.spec}
+
+##########################################
+# other global stuff
+
+function main {
+    gen_tar
+    gen_spec
+    if [[ "${GITHUB_ACTIONS:-}" != "true" ]]
+    then
+        # GitHub Actions calls this script directly
+        make CHANGES="${tmp_dir}/skuba.changes.append" suse-changelog
+
+        # Assemble for OBS / manual use later 
+        rm -rf "${rpm_files}"
+        mkdir -p "${rpm_files}"
+        cp "${tmp_dir}"/* "${rpm_files}"
+        log "Find files for RPM package in ${rpm_files}"
+
+        ibs_user=$(osc config https://api.suse.de user | awk '$0=$NF' \
+                   || echo -n '')
+        if [[ -n "$ibs_user" ]]
+        then
+            do_ibs_update "$ibs_user"
```

Comment:
I could see pulling this logic into a separate script that Jenkins runs.  I wanted to leave it in script which could be run locally as well, though, so that the command-line process would work the same way as an automatic Jenkins execution.  Imagine, for example, that for some reason we wanted to release skuba with a previous tag using a plus version string or similar.

I think what I'd like to do is to handle that change in the same PR as creation of the Jenkins job, though.  It makes sense to me if we bundle all of the Jenkins-relevant changes into one "automate release with Jenkins" PR.

Reply:
With the Github Action, however, I'll note that the Jenkins flow is quite a bit simpler.  Because the Action creates a release with an artifact containg the OBS files, no checkout is necessary. All the Jenkens script needs to do is:

1. get OBS tarball from the current skuba release (see https://github.com/SUSE/skuba/pull/1371#issuecomment-701043136 above)
1. extract tarball
1. prepend changelog from tarball to current changelog
1. update non-changelog files in OBS repo direct from release tarball
1. run services to update vendored modules
1. commit
1. create SR (maybe)

I've been looking at the details of the OBS services we have, and several of those steps might be doable via _service directly; what I'd *really* like to do is be able to just hit the "run service" endpoint in IBS and have it do the download, extract, and changelog update.  But we might need to do the download and changelog update separately, and then merely let obs do the extraction & vendoring

This is more of a discussion for the Jenkinsfile PR, though.  I'll get that opened up in a sec and move this over there.

Reply:
Jenkins PR is #1395